src/lib/client.ts

```diff
@@ -65,8 +65,19 @@ export class HTTPClient implements IRequestClient {
 
         axiosInstance.interceptors.response.use(
             (response) => response,
-            // eslint-disable-next-line @typescript-eslint/promise-function-async
-            (error) => Promise.reject(error.response.data),
+            // eslint-disable-next-line @typescript-eslint/promise-function-async,@typescript-eslint/tslint/config
+            (error) => {
+                if (!!error.response) {
+                    return Promise.reject(error.response.data);
+                } else if (!!error.request) {
+                    return Promise.reject(error.request);
+                } else {
+                    return Promise.reject({
+                        message: error.message,
+                        code: error.code,
+                    });
```

Comment:
The [Axios docs](https://github.com/axios/axios#handling-errors) say this state occurs when: _Something happened in setting up the request that triggered an Error_. Is it guaranteed the `error` object always has a `code` key in this case? Or do we need to do some checking for this?

Reply:
According to [Axios docs](https://github.com/axios/axios#interceptors) regarding response interceptors, it is stated that the `onRejected` functionality (which is the block the above code is part of), it is triggered for any status code that falls outside the range of 2xx. In our case, the above code is used to handle all other errors that do not have either a request or a response (for instance a DNS error). Taking these into account, the `error` object will always have a `code` key in this case.